<a href="https://colab.research.google.com/github/wmhowarth/DataSets-Project2/blob/main/Project2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2

## Part 1

### Author : Warren Howarth

In [2]:
# import libraries
import pandas as pd

- Dataset 1
-- Stroke Prediction DataSet

In [3]:
# import data
fname = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/healthcare-dataset-stroke-data.csv'
df = pd.read_csv(fname)
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [4]:
# Get an idea of the data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


1. Source of data
- This data is found on kaggle
https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

2. Brief description of data
- This data is used to predict if a patient is likely to have a stroke based on certain characteristics and parameters of the patient

3. What is the target?
- Will the patient have a stroke (1) or not (0) based on the value in the 'stroke' column

4. What does one row represent? (A person? A business? An event? A product?)
- A patient

5. Is this a classification or regression problem?
- Classification

6. How many features does the data have?
- I would drop 'id', so 10

7. How many rows are in the dataset?
- 5110

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- The 'bmi' feature is missing values, it will be difficult determining what to do with these rows
- There are quite a few categorical features that have a number of classes, making the total number of features once the data is processed quite large

- Dataset 2
-- Spanish Wine Quality Dataset

In [5]:
# import data
fname = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/wines_SPA.csv'
df = pd.read_csv(fname)
df.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

In [6]:
# Get an idea of the data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


1. Source of data
- This data is found on kaggle
https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset

2. Brief description of data
- This data is used to predict either the rating or the price of red wines in Spain based on a number of characteristics of the wine

3. What is the target?
- Either the 'rating' from 1 - 5, or the 'price' of the wine

4. What does one row represent? (A person? A business? An event? A product?)
- A red wine variety in Spain

5. Is this a classification or regression problem?
- It can be both a Classification (rating) and Regression (price) problem

6. How many features does the data have?
- 10

7. How many rows are in the dataset?
- 7500

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- The 'year', 'type', 'body' and 'acidity' features are all missing values, it will be difficult determining what to do with this missing data
- If I decide to use the Classification problem and predict the rating, it is not a simple true/false scenario